### DISCUSSION TITLE
Bulit in API for "advanced" parallel iteration

### DISCUSSION THREAD

**Discussion by dumblob** [vlang/v#11706] [Ideas]
Separately from built-in parallelism & concurrency (i.e. backend) is the topic of APIs (i.e. programmer-facing "front end").

Hereby I'd like to ask for interest in parallel iteration API like e.g. [Rayon](https://github.com/rayon-rs/rayon ) uses (see also http://smallcultfollowing.com/babysteps/blog/2017/04/06/rayon-0-7-released/ ), but **built in** to standard *V* library (imagine e.g. parallel iteration over **any** array). Of course, such API **must** be seamlessly compatible (i.e. complement & enhance) all the built-in parallelism primitives (including go routines & channels).

Note, this might complement (or even conflict) with iteration approaches e.g. [Weave](https://github.com/mratsim/weave/ ) uses. So keep that in mind and take a thorough look first.

Just for clarification, I don't want any syntactic support, but just API.

**Comment by medvednikov** [comment]
Yeah I had the same idea.
